### PR TITLE
Removes the hardcoded path for input.nml

### LINKED ISSUE #779: relax the restriction on hard-wired references to `input.nml`
**Is your feature request related to a problem? Please describe.**
No.  Not a problem but poses a limitation.

**Describe the solution you'd like**
`input.nml` is hard-wired and expected to exist in CWD.
For applications in DA with outer loops, this poses a limitation when there is a need to read a unique input.nml for that outer loop.

**Describe alternatives you've considered**
Moving `input.nml` in compiled code that uses FMS.

### PR DESCRIPTION
**Pull request by GFDL-Eric**
**Description**
The path for the input namelist can now be specified in the fms_init call. All references to input.nml have been replaced by a reference to either the path (in the case of ingesting the file contents) or the input_nml_file variable which contains the namelist information. References using '#ifdef INTERNAL_FILE_NML' have been removed since this is now always defined.

Fixes #779 

**How Has This Been Tested?**
Compiles with Intel19 on Skylake.

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: mpp/mpp.F90 [hunk 1]
```diff
@@ -1280,7 +1280,7 @@ module mpp_mod
 !
 ! parameter defining length of character variables
   integer, parameter :: INPUT_STR_LENGTH = 256
```
Comment (bensonr) [inline mpp/mpp.F90:1282]: I don't think we need to define or publicize this parameter anymore as it is no longer used within our mpp ascii routines.  I found the only use is now within our test harness.

Comment (GFDL-Eric) [reply mpp/mpp.F90]: Fixed in latest push

**Review by bensonr — APPROVED** [review]

File: mpp/include/mpp_comm_mpi.inc [hunk 2]
```diff
@@ -111,27 +112,17 @@
...
-  end do
-
-  open(unit_nml,file='input.nml', iostat=io_status)
-  read(unit_nml,mpp_nml,iostat=io_status)
-  close(unit_nml)
-#endif
-
   if (io_status > 0) then
-     call mpp_error(FATAL,'=>mpp_init: Error reading input.nml')
+     call mpp_error(FATAL,'=>mpp_init: Error reading mpp_nml')
```
Comment (thomas-robinson) [inline mpp/include/mpp_comm_mpi.inc:125]: This is not descriptive and probably needs to be updated

**Review by rem1776 — APPROVED** [review]

File: mpp/include/mpp_comm_nocomm.inc [hunk 3]
```diff
@@ -86,26 +87,16 @@ subroutine mpp_init( flags,localcomm )
...
-  end do
-
-  open(unit_nml,file='input.nml', iostat=io_status)
-  read(unit_nml,mpp_nml,iostat=io_status)
-  close(unit_nml)
-#endif
-
   if (io_status > 0) then
-     call mpp_error(FATAL,'=>mpp_init: Error reading input.nml')
+     call mpp_error(FATAL,'=>mpp_init: Error reading mpp_nml')
```
Comment (thomas-robinson) [inline mpp/include/mpp_comm_nocomm.inc:99]: This is not descriptive and probably needs to be updated

**Review by thomas-robinson — APPROVED** [review]

**Pull request merged**